### PR TITLE
 Migrate to es6 Promise. Add fetchToken support.

### LINKED ISSUE #57: Token based authentication is broken in Safari
There appears to be a feature (bug?) in the latest Safari (v12.1) where an attempt to resume AudioContext upon completion of an async operation fails, even it was initiated by a user action.
This makes it impossible to effectively use AuthorizationToken, as it only makes sense to fetch one upon a user action (e.g. button click) and the fetch has to be asynchronous. Plus there is no way to wait till AudioContext is resumed before supplying the token.

Proposed solution: CognitiveSubscriptionKeyAuthentication class already supports fetch (and fetchOnExpiry) function as a parameter. If it could be passed in via SpeechConfig, the issue would be solved. One obvious obstacle - fetch function returns proprietary Promise. Hence the part of the solution has to be the migration to es6 Promise.

### PR DESCRIPTION
**Pull request by Domratchev**
Resolves #57

### CODE REVIEW

**Comment by rhurey** [comment]
First, thanks. Moving to ES6 Promises and getting the ability to write async / await code has been something on the "want" list for quite a while.

I'd started down migrating to native promises back in March when #39 was filed, and hadn't pushed the branch yet.

There were a few reasons that I held off over:
1) I wanted to be able to push the change all the way through the API surface and change to use Promise instead of the continuation pattern. But the time wasn't right for a change like that. (We want to keep the various SDK language bindings aligned in versioning and a breaking API surface change would require us to go to v2.0)

2) The risk of breaking something in the implementation seemed too high for a minor version release.

I'll take an in-depth look through the changes in the next couple of days, but my initial comments are that we can't move the property access methods around on the config objects.

Also, some behavior difference exists between the Promise implementations as there's a good sized chunk of tests failing now.

File: src/sdk/SpeechTranslationConfig.ts [hunk 1]
```diff
@@ -92,26 +91,6 @@ export abstract class SpeechTranslationConfig extends SpeechConfig {
...
-    public abstract set authorizationToken(value: string);
-
-    /**
-     * Gets/Sets the speech recognition language.
-     * @member SpeechTranslationConfig.prototype.speechRecognitionLanguage
-     * @function
-     * @public
-     * @param {string} value - The authorization token.
-     */
-    public abstract set speechRecognitionLanguage(value: string);
```
Comment (rhurey) [inline src/sdk/SpeechTranslationConfig.ts]: We can't remove methods from the SDK API surface without a major version revision, and we're not ready to move the SDK to 2.0 yet, so these should stay.

File: src/sdk/SpeechTranslationConfig.ts [hunk 2]
```diff
@@ -147,24 +126,6 @@ export abstract class SpeechTranslationConfig extends SpeechConfig {
...
-     */
-    public abstract setProperty(name: string, value: string): void;
-
-    /**
-     * Dispose of associated resources.
-     * @member SpeechTranslationConfig.prototype.close
-     * @function
-     * @public
-     */
-    public abstract close(): void;
```
Comment (rhurey) [inline src/sdk/SpeechTranslationConfig.ts]: Same comment as above about removing API members.

File: src/sdk/SpeechTranslationConfig.ts (hunk 1 above)
Comment (Domratchev) [reply src/sdk/SpeechTranslationConfig.ts]: SpeechTranslationConfig extends SpeechConfig which already declares/defines these properties and methods.

File: src/sdk/SpeechTranslationConfig.ts (hunk 2 above)
Comment (Domratchev) [reply src/sdk/SpeechTranslationConfig.ts]: SpeechTranslationConfig extends SpeechConfig which already declares/defines these properties and methods.
There is no change in the exposed API.

**Comment by Domratchev** [comment]
@rhurey Ryan, do you see the migration to ES6 promises happening any time soon? I don't think that authorization token issue in Safari has any other solution.

**Comment by orgads** [comment]
Can this be closed now?

**Comment by glharper** [comment]
Closing due to inactivity.

**Pull request closed without merging**